### PR TITLE
Am testing

### LINKED ISSUE #1: Homepage: View
The user should be able to see: 

- [ ] a search bar to search for books

- [ ] a nav bar

- [ ]  a side bar

- [ ] a recent list of best selling books/their list of books to read

### PR DESCRIPTION
**Pull request by aemcdonald**
### Description
Created GoalForm class component with controlled form for user input
Created user interface to allow a user to set a goal
Added action & reducer to set the user goal in state
Rearranged files to organize components directory & maintain consistent modular file structure
Added functionality to display "to read" & "have read" list when the user navigates to the appropriate page
Set up test files with initial imports & boilerplate test to ensure that files properly hooked up
Add conditional render to Have Read & To Read buttons

### Types of changes:
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Has This Been Tested?
- [x] No
- [ ] Yes
No tests written yet, I tested my changes manually in the browser.

### What are the relevant tickets?
closes #1 #12 #27 

### Checklist:
- [x] My code follows the code style of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/Components/GoalForm/GoalForm.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import './GoalForm.css';
+
+class GoalForm extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  handleUserInput = (event) => {
+    this.setState({ userGoal: event.target.value })
+  }
```
Comment (stacyp2006) [inline src/Components/GoalForm/GoalForm.js:13]: We will need a method to clear the input after a user has set a goal. I think we already have an issue for this. We may also want to do some conditional rendering to have the submit button change text from "submit" to "update" so a user can update their goal.

File: src/Components/BookCard/BookCard.js [hunk 2]
```diff
@@ -12,12 +13,23 @@ const BookCard = ({ bookInfo, addToToReadList, addToHaveReadList } ) => {
           <h5 className='rank'>Rank: {bookInfo.rank}</h5>
         </div>
         <div className='book-btns'>
-          <button className='to-read-btn' type='button' value={bookInfo.title} onClick={event => addToToReadList(event)}>To Read</button>
-          <button className='have-read-btn' type='button' value={bookInfo.title} onClick={event => addToHaveReadList(event)}>Have Read</button>
+          {!toReadList.includes(bookInfo) &&
```
Comment (nware1066) [inline src/Components/BookCard/BookCard.js:16]: definitely something to add as a future issue, when we are not concerned about the time constraints. Either just having the array reject duplicates, or having the buttons only render if the book is not in a list. Adding a Boolean in state to indicate if a book is read, or is a favorite might be a good solution to the button rendering issue

File: src/Components/BookCard/BookCard.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import BookCard from './BookCard';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+describe('BookCard', () => {
```
Comment (nware1066) [inline src/Components/BookCard/BookCard.test.js:6]: good to be sure this is set up and working, definitely want to make this more robust

File: src/Components/GoalForm/GoalForm.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  setUserGoal = (event) => {
+    this.props.createUserGoal(this.state.userGoal)
+  }
+
+  render() {
+    return(
+      <form>
+        <input
+          type='number'
```
Comment (stacyp2006) [inline src/Components/GoalForm/GoalForm.js:23]: Let's add some text here that says something like, "Your reading goal is 25 books!"

**Review by stacyp2006 — APPROVED** [review]

File: src/Components/GoalForm/GoalForm.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { createUserGoal } from '../../actions';
+import './GoalForm.css';
+
+class GoalForm extends Component {
+  constructor(props) {
+    super(props);
```
Comment (nware1066) [inline src/Components/GoalForm/GoalForm.js:8]: good to have the goal form here. This is what we have decided to put on the sidebar, I believe

**Review by nware1066 — APPROVED** [review]

**Pull request merged**